crates/networking/rpc/engine/blobs.rs

```diff
@@ -110,55 +115,83 @@ impl RpcHandler for BlobsV2Request {
 
     async fn handle(&self, context: RpcApiContext) -> Result<Value, RpcErr> {
         info!("Received new engine request: Requested Blobs V2");
-        if self.blob_versioned_hashes.len() >= GET_BLOBS_V1_REQUEST_MAX_SIZE {
-            return Err(RpcErr::TooLargeRequest);
+        let res = get_blobs_and_proof_v2(&self.blob_versioned_hashes, context).await?;
+        if res.iter().any(|blob| blob.is_none()) {
+            return Ok(Value::Null);
         }
+        serde_json::to_value(res).map_err(|error| RpcErr::Internal(error.to_string()))
+    }
+}
 
-        if let Some(current_block_header) = context
-            .storage
-            .get_block_header(context.storage.get_latest_block_number().await?)?
-            && !context
-                .storage
-                .get_chain_config()
-                .is_osaka_activated(current_block_header.timestamp)
-        {
-            // validation requested in https://github.com/ethereum/execution-apis/blob/a1d95fb555cd91efb3e0d6555e4ab556d9f5dd06/src/engine/osaka.md?plain=1#L130
-            return Err(RpcErr::UnsuportedFork(
-                "getBlobsV2 engine only supported for Osaka".to_string(),
-            ));
+impl RpcHandler for BlobsV3Request {
+    fn parse(params: &Option<Vec<Value>>) -> Result<Self, RpcErr> {
+        let params = params
+            .as_ref()
+            .ok_or(RpcErr::BadParams("No params provided".to_owned()))?;
+        if params.len() != 1 {
+            return Err(RpcErr::BadParams("Expected 1 param".to_owned()));
         };
+        Ok(BlobsV3Request {
+            blob_versioned_hashes: serde_json::from_value(params[0].clone())?,
+        })
+    }
 
-        let mut res: Vec<Option<BlobAndProofV2>> = vec![None; self.blob_versioned_hashes.len()];
+    async fn handle(&self, context: RpcApiContext) -> Result<Value, RpcErr> {
+        info!("Received new engine request: Requested Blobs V3");
+        let res = get_blobs_and_proof_v2(&self.blob_versioned_hashes, context).await?;
+        serde_json::to_value(res).map_err(|error| RpcErr::Internal(error.to_string()))
+    }
+}
 
-        for blobs_bundle in context.blockchain.mempool.get_blobs_bundle_pool()? {
-            // Go over all blobs bundles from the blobs bundle pool.
-            let blobs_in_bundle = blobs_bundle.blobs;
-            let commitments_in_bundle = blobs_bundle.commitments;
-            let proofs_in_bundle = blobs_bundle.proofs;
+// Get blob data for a given list of blob versioned hashes.
```

Comment:
It's better to use a rustdoc here (`///`)

```suggestion
/// Get blob data and proofs for a given list of blob versioned hashes.
```